Kernel 4.11.9 does not compile...

Hi

After upgrading to kernel 4.11.9 in Arch Linux i get this compile error:

/var/lib/dkms/rtl8812au_rtl8821au/4.3.20.r16.d716b38/build/os_dep/linux/ioctl_cfg80211.c: In function ‘rtw_cfg80211_add_monitor_if’:
/var/lib/dkms/rtl8812au_rtl8821au/4.3.20.r16.d716b38/build/os_dep/linux/ioctl_cfg80211.c:3952:12: error: ‘struct net_device’ has no member named ‘destructor’; did you mean ‘priv_destructor’?
  mon_ndev->destructor = rtw_ndev_destructor;
            ^~~~~~~~~~
            priv_destructor
make[2]: *** [scripts/Makefile.build:295: /var/lib/dkms/rtl8812au_rtl8821au/4.3.20.r16.d716b38/build/os_dep/linux/ioctl_cfg80211.o] Error 1
make[1]: *** [Makefile:1492: _module_/var/lib/dkms/rtl8812au_rtl8821au/4.3.20.r16.d716b38/build] Error 2
make[1]: Leaving directory '/usr/lib/modules/4.11.9-1-ARCH/build'
make: *** [Makefile:1670: modules] Error 2

How can i fix this ? Is it really so simple in renaming the destructor to priv_destructor?

Obviously the module does not build and so no wifi connections work.

Thanks!

--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -3336,7 +3336,12 @@ static int rtw_cfg80211_add_monitor_if(_adapter *padapter, char *name, struct ne
 	mon_ndev->type = ARPHRD_IEEE80211_RADIOTAP;
 	strncpy(mon_ndev->name, name, IFNAMSIZ);
 	mon_ndev->name[IFNAMSIZ - 1] = 0;
-	mon_ndev->destructor = rtw_ndev_destructor;
+#if (LINUX_VERSION_CODE>=KERNEL_VERSION(4,11,9))
+	mon_ndev->needs_free_netdev = false;
+	mon_ndev->priv_destructor = rtw_ndev_destructor;
+#else
+  	mon_ndev->destructor = rtw_ndev_destructor;
+#endif
 	
 #if (LINUX_VERSION_CODE>=KERNEL_VERSION(2,6,29))
 	mon_ndev->netdev_ops = &rtw_cfg80211_monitor_if_ops;

